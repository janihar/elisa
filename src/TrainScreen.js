import React, { useState, useEffect } from "react";
import { StyleSheet, View, Dimensions, FlatList, Text } from "react-native";
import Search from "./Search";
import TrainItem from "./TrainItem";

const TrainScreen = ({ navigation }) => {
  //Getting json from server which includes every latest train numbers
  const [trainNumbers, setTrainNumbers] = useState();
  const [didFound, setDidFound] = useState(true);

  //Fetch every latest trains
  const fetchTrains = async () => {
    const train = await fetch(
      "https://rata.digitraffic.fi/api/v1/train-locations/latest/"
    );
    const trainData = await train.json();
    setTrainNumbers(trainData);
  };

  //Specific train fetch
  const fetchSpecificTrain = async trainID => {
    //Fetching every latest train
    const train = await fetch(
      "https://rata.digitraffic.fi/api/v1/train-locations/latest/" + trainID
    );
    const trainData = await train.json();
    if (trainData.length === 0) {
      // If nothing is found then set DidFound false and show message to user
      setDidFound(false);
      setTrainNumbers(trainData);
    } else {
      //Train did found, no error message needed
      setDidFound(true);
      setTrainNumbers(trainData);
    }
  };

  //Component mounts for the very first time
  useEffect(() => {
    fetchTrains();
  }, []);

  return (
    <View style={styles.container}>
      <Search fetchTrain={fetchSpecificTrain} navigation={navigation} />

      {didFound === false && <Text>JUNAA EI LÃ–YTYNYT</Text>}

      <FlatList
        style={styles.list}
        data={trainNumbers}
        renderItem={({ item }) => <TrainItem trainNumber={item.trainNumber} navigation={navigation} />}
        keyExtractor={item => item.trainNumber.toString()}
      />
    </View>
  );
};

export default TrainScreen;

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: "#fff",
    alignItems: "center",
    justifyContent: "center"
  },
  list: {
    flex: 1,
    width: Dimensions.get("window").width / 1.5,
    marginHorizontal: 16,
    top: Dimensions.get("window").height / 10
  },
  mapStyle: {
    width: Dimensions.get("window").width,
    height: Dimensions.get("window").height,
    top: 100
  }
});
