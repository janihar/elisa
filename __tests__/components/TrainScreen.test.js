import React from "react";
import renderer from "react-test-renderer";
import TrainScreen from "../../src/TrainScreen";

describe("<TrainScreen /> ", () => {

  it("renders correctly", () => {
    const tree = renderer.create(<TrainScreen />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});