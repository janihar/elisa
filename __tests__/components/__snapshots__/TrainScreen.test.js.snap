// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TrainScreen />  renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#fff",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#fff",
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <Image
      source={1}
      style={
        Object {
          "flex": 1,
          "height": 50,
          "resizeMode": "contain",
          "width": 200,
        }
      }
    />
    <TextInput
      allowFontScaling={true}
      onChangeText={[Function]}
      placeholder="Syötä juna"
      rejectResponderTermination={true}
      style={
        Object {
          "borderColor": "gray",
          "borderRadius": 20,
          "borderWidth": 1,
          "bottom": 20,
          "height": 40,
          "textAlign": "center",
          "width": 500,
        }
      }
      underlineColorAndroid="transparent"
    />
    <View
      style={
        Object {
          "borderRadius": 25,
          "color": "#64B82C",
          "fontSize": 20,
          "width": 375,
        }
      }
    >
      <View
        accessibilityRole="button"
        accessibilityStates={Array []}
        accessible={true}
        focusable={true}
        isTVSelectable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Array [
              Object {},
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#007AFF",
                  "fontSize": 18,
                  "padding": 8,
                  "textAlign": "center",
                },
                Object {
                  "color": "#64B82C",
                },
              ]
            }
          >
            HAE
          </Text>
        </View>
      </View>
    </View>
  </View>
  <View>
    <ActivityIndicator
      animating={true}
      color="#999999"
      hidesWhenStopped={true}
      size="large"
    />
    <Text
      style={
        Object {
          "alignItems": "center",
          "textAlign": "center",
        }
      }
    >
      Haetaan
    </Text>
  </View>
</View>
`;
